pkg/time/time_skew_amd64.go

```diff
@@ -150,7 +154,154 @@ func (it *TimeSkew) Inject(pid int) error {
+		}
+	}()
+
+	var vdsoEntry *mapreader.Entry
+	for index := range program.Entries {
+		// reverse loop is faster
+		e := program.Entries[len(program.Entries)-index-1]
+		if e.Path == vdsoEntryName {
+			vdsoEntry = &e
+			break
+		}
+	}
+	if vdsoEntry == nil {
+		return errors.Errorf("cannot find [vdso] entry, pid: %d", pid)
+	}
+
+	// minus tailing variable part
+	// every variable has 8 bytes
+	constImageLen := len(it.fakeImage.content) - 8*len(it.fakeImage.offset)
+	var fakeEntry *mapreader.Entry
+
+	// find injected image to avoid redundant inject (which will lead to memory leak)
+	for _, e := range program.Entries {
+		e := e
+
+		image, err := program.ReadSlice(e.StartAddress, uint64(constImageLen))
+		if err != nil {
+			continue
+		}
+
+		if bytes.Equal(*image, it.fakeImage.content[0:constImageLen]) {
+			fakeEntry = &e
+			log.Info("found injected image", "addr", fakeEntry.StartAddress, "pid", pid)
+			break
+		}
+	}
+
+	// target process has not been injected yet
+	if fakeEntry == nil {
+		fakeEntry, err = program.MmapSlice(it.fakeImage.content)
+		if err != nil {
+			return errors.Wrapf(err, "mmap fake image, pid: %d", pid)
+		}
+
+		originAddr, err := program.FindSymbolInEntry(getTimeOfDay, vdsoEntry)
+		if err != nil {
+			return errors.Wrapf(err, "find origin gettimeofday in vdso, pid: %d", pid)
+		}
+
+		err = program.JumpToFakeFunc(originAddr, fakeEntry.StartAddress)
+		if err != nil {
+			return errors.Wrapf(err, "override origin gettimeofday, pid: %d", pid)
+		}
+	}
+
+	err = program.WriteUint64ToAddr(fakeEntry.StartAddress+uint64(it.fakeImage.offset[externVarTvSecDelta]), uint64(it.deltaSeconds))
+	if err != nil {
+		return errors.Wrapf(err, "set %s for time skew, pid: %d", externVarTvSecDelta, pid)
+	}
+
+	err = program.WriteUint64ToAddr(fakeEntry.StartAddress+uint64(it.fakeImage.offset[externVarTvNsecDelta]), uint64(it.deltaNanoSeconds))
+	if err != nil {
+		return errors.Wrapf(err, "set %s for time skew, pid: %d", externVarTvNsecDelta, pid)
+	}
+	return nil
+}
+
+func (it *SkewGetTimeOfDay) Recover(pid int) error {
+	zeroSkew := NewTimeOfDaySkewWithCustomFakeImage(0, 0, it.fakeImage)
+	return zeroSkew.Inject(pid)
+}
+
+func (it *CompositeInjector) Inject(pid int) error {
+	for _, injector := range it.injectors {
+		if err := injector.Inject(pid); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (it *CompositeInjector) Recover(pid int) error {
+	for _, injector := range it.injectors {
+		if err := injector.Recover(pid); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func NewCompositeInjector(deltaSec int64, deltaNsec int64, clockIdsMask uint64) (*CompositeInjector, error) {
+	SkewClockGetTime, err := NewTimeSkew(deltaSec, deltaNsec, clockIdsMask)
+	if err != nil {
+		return nil, err
+	}
+	SkewGetTimeOfDay, err := NewTimeOfDaySkew(deltaSec, deltaNsec)
+	if err != nil {
+		return nil, err
+	}
+	return &CompositeInjector{injectors: []FakeClockInjector{SkewClockGetTime, SkewGetTimeOfDay}}, nil
+}
```

Comment:
```suggestion
func NewTimeSkew(deltaSec int64, deltaNsec int64, clockIdsMask uint64) (FakeClockInjector, error) {
	SkewClockGetTime, err := NewTimeSkew(deltaSec, deltaNsec, clockIdsMask)
	if err != nil {
		return nil, err
	}
	SkewGetTimeOfDay, err := NewTimeOfDaySkew(deltaSec, deltaNsec)
	if err != nil {
		return nil, err
	}
	return &CompositeInjector{injectors: []FakeClockInjector{SkewClockGetTime, SkewGetTimeOfDay}}, nil
}
```


Because of the parameter, it is not the constructor of `CompositeInjector`, use another meaningful name.